pkg/autoscaler/autoscaler.go

```diff
@@ -732,6 +518,95 @@ var _ = Describe("[Feature:Machines] Autoscaler should", func() {
+			machineSetParams.Labels[targetedNodeLabel] = ""
+			transientMachineSet, err = framework.CreateMachineSet(client, machineSetParams)
+			Expect(err).ToNot(HaveOccurred())
+			cleanupObjects[transientMachineSet.GetName()] = transientMachineSet
+
+			By("Waiting for all Machines in the MachineSet to enter Running phase")
+			framework.WaitForMachineSet(client, transientMachineSet.GetName())
+
+			// To exercise the MaxNodesTotal mechanism we want to make sure
+			// that the MachineSet can grow large enough to reach the boundary.
+			// A simple way to test this is by setting the max scale size to
+			// the MaxNodesTotal+1, since we will not be able to reach this limit
+			// due to the original install master/worker nodes.
+			maxMachineSetReplicas := int32(caMaxNodesTotal + 1)
+			By(fmt.Sprintf("Creating a MachineAutoscaler backed by MachineSet %s - min: 1, max: %d",
+				transientMachineSet.GetName(), maxMachineSetReplicas))
+			asr := machineAutoscalerResource(transientMachineSet, 1, maxMachineSetReplicas)
+			Expect(client.Create(ctx, asr)).Should(Succeed())
+			cleanupObjects[asr.GetName()] = asr
+
+			// We want to create a workload that would cause the autoscaler to
+			// grow the cluster beyond the MaxNodesTotal. If we set the replicas
+			// to the maximum MachineSet size this will create enough demand to
+			// grow the cluster to maximum size.
+			jobReplicas := maxMachineSetReplicas
+			By(fmt.Sprintf("Creating scale-out workload: jobs: %v, memory: %s",
+				jobReplicas, workloadMemRequest.String()))
+			workload := newWorkLoad(jobReplicas, workloadMemRequest, targetedNodeLabel)
+			cleanupObjects[workload.GetName()] = workload
+			Expect(client.Create(ctx, workload)).Should(Succeed())
+
+			// At this point the autoscaler should be growing the cluster, we
+			// wait until the cluster has grown to reach MaxNodesTotal size.
+			By(fmt.Sprintf("Waiting for cluster to scale up to %d nodes", caMaxNodesTotal))
+			Eventually(func() (bool, error) {
+				nodes, err := framework.GetNodes(client)
+				if err != nil {
+					return false, err
+				}
+				if len(nodes) != caMaxNodesTotal {
+					return false, nil
+				}
+				return true, nil
+			}, framework.WaitLong, pollingInterval).Should(BeTrue())
+
+			// Wait for all nodes to become ready, we wait here to help ensure
+			// that the cluster has reached a steady state and no more machines
+			// are in the process of being added.
+			By("Waiting for all Machines in MachineSet to enter Running phase")
+			framework.WaitForMachineSet(client, transientMachineSet.GetName())
+
+			// Now that the cluster has reached maximum size, we want to ensure
+			// that it doesn't try to grow larger.
+			By("Watching Cluster node count to ensure it remains consistent")
+			Consistently(func() (bool, error) {
+				nodes, err := framework.GetNodes(client)
+				if err != nil {
+					return false, err
+				}
+				if len(nodes) != caMaxNodesTotal {
+					return false, nil
+				}
+				return true, nil
```

Comment:
You could just do this here:
```golang
nodes, err := framework.GetNodes(client)
return len(nodes) == caMaxNodesTotal, err
``` 
It will assert the first value is true, and all the following values are `nil`:
https://github.com/onsi/gomega/blob/master/gomega_dsl.go#L193-L195